magma/lib/magma/actions/add_project.rb

```diff
@@ -37,6 +38,60 @@ def validate_project_name
 
     private
 
+    def setup_metis
+      # Create a Metis bucket, owned by Magma,
+      #   for linking files into.
+      host = Magma.instance.config(:storage).fetch(:host)
+
+      client = Etna::Client.new("https://#{host}", @user.token)
+
+      bucket_create_route = client.routes.find { |r| r[:name] == 'bucket_create' }
+
+      begin
+        @errors << Magma::ActionError.new(
+          message: "No bucket_create route on the Metis storage host -- will not be able to link files for this",
+          source: 'setup_metis'
+        )
+        return nil
+      end unless bucket_create_route
+
+      path = client.route_path(
+        bucket_create_route,
+        project_name: @project_name,
+        bucket_name: 'magma'
+      )
+
+      params = {
+        owner: 'magma',
+        description: 'For magma use only',
+        access: 'administrator'
+      }
+
+      # Now populate the standard headers
+      hmac_params = {
```

Comment:
Hmm, looking at the metis bucket#create endpoint it does not appear to make any checks whatsoever that the create request is signed by the assigned owner (i.e., i can create a bucket with owner: 'magma' without Magma signing the request). Dunno if we should resolve this by fixing the endpoint or by removing this hmac param here.

Reply:
Thinking on this a bit more, there are a couple of other checks that we suggested at one point that don't exist on `bucket#create` - checking that someone else doesn't make the `magma` bucket and set its owner to something other than `magma`. Probably the right thing to do here is to change `bucket#create` so it won't allow you to create a bucket with `@params[:bucket_name] == @params[:owner]` unless it is signed by `@params[:owner]`